File: public/developer-downloads/examples/java/Quickstart.java
import com.fasterxml.jackson.databind.JsonNode;

// Java 17+. Reuse AureviaProvider from your authenticated backend controller.
public final class Quickstart {
    public static JsonNode createPlayerLaunch(AureviaProvider provider, String authenticatedPlayerId,
        String persistedLaunchKey, String approvedReturnUrl) throws Exception {
        var catalog = provider.request("GET", "api/provider/v1/games?provider_id=playngo", null, null);
        var games = catalog.required("items");
        if (games.isEmpty()) throw new IllegalStateException("No games assigned.");
        var session = provider.launch(authenticatedPlayerId, approvedReturnUrl, persistedLaunchKey,
            games.get(0).required("symbol").asText(), "desktop");
        // Store session_id with this player. Return only launch_url to their browser.
        // Never log launch URLs or expose provider credentials to the frontend.
        return session;
    }
}

// Configure the reusable client with your server's secret store:
// new AureviaProvider(PROVIDER_BASE_URL, OPERATOR_API_KEY)
// Reconcile: provider.request("GET", "api/provider/v1/transactions?limit=50", null, null)
// Close: provider.request("POST", "api/provider/v1/sessions/" +
//     java.net.URLEncoder.encode(storedSessionId, java.nio.charset.StandardCharsets.UTF_8)
//       .replace("+", "%20") + "/close", null, null)
// A timed-out launch must reuse the SAME persisted key AND request body.

File: public/developer-downloads/examples/quickstart.mjs
// Node.js 20+. Call this from your authenticated backend route.
import {providerRequest, launchGame} from './provider.mjs';

export async function createPlayerLaunch(authenticatedPlayerId, persistedLaunchKey, approvedReturnUrl) {
  const base = process.env.PROVIDER_BASE_URL;
  const key = process.env.OPERATOR_API_KEY;
  if (!base || !key) throw new Error('Configure server-side provider credentials');
  const catalog = await providerRequest(base, key, 'GET', 'api/provider/v1/games?provider_id=playngo');
  if (!catalog.items.length) throw new Error('No games assigned');
  const session = await launchGame(base, key, authenticatedPlayerId, approvedReturnUrl,
    persistedLaunchKey, catalog.items[0].symbol, 'desktop');
  // Store session.session_id with this player for lookup and close.
  // Return ONLY session.launch_url to their authenticated browser; do not log it.
  return session;
}

// Reconcile: providerRequest(base, key, 'GET', 'api/provider/v1/transactions?limit=50');
// Close: providerRequest(base, key, 'POST',
//   `api/provider/v1/sessions/${encodeURIComponent(storedSessionId)}/close`);
// A timed-out launch must reuse the SAME persisted key AND request body.
